driver/src/main/com/mongodb/client/gridfs/GridFSBucket.java

```diff
@@ -0,0 +1,253 @@
+     *
+     * @param id the ObjectId of the file to be written to the destination stream
+     * @param destination the destination stream
+     */
+    void downloadToStream(ObjectId id, OutputStream destination);
+
+    /**
+     * Opens a Stream from which the application can read the contents of the stored file specified by {@code id}.
+     *
+     * @param id the custom id value of the file, to be put into a stream.
+     * @return the stream
+     * @deprecated using custom id values for with GridFS is no longer supported
+     */
+    @Deprecated
+    GridFSDownloadStream openDownloadStream(BsonValue id);
+
+    /**
+     * Downloads the contents of the stored file specified by {@code id} and writes the contents to the {@code destination} Stream.
+     *
+     * @param id the custom id of the file, to be written to the destination stream
+     * @param destination the destination stream
+     * @deprecated using custom id values for with GridFS is no longer supported
+     */
+    @Deprecated
+    void downloadToStream(BsonValue id, OutputStream destination);
+
+    /**
+     * Opens a Stream from which the application can read the contents of the latest version of the stored file specified by the
+     * {@code filename}.
+     *
+     * @param filename the name of the file to be downloaded
+     * @return the stream
+     */
+    GridFSDownloadStream openDownloadStreamByName(String filename);
+
+    /**
+     * Opens a Stream from which the application can read the contents of the stored file specified by {@code filename} and the revision
+     * in {@code options}.
+     *
+     * @param filename the name of the file to be downloaded
+     * @param options the download options
+     * @return the stream
+     */
+    GridFSDownloadStream openDownloadStreamByName(String filename, GridFSDownloadByNameOptions options);
+
+    /**
+     * Downloads the contents of the latest version of the stored file specified by {@code filename} and writes the contents to
+     * the {@code destination} Stream.
+     *
+     * @param filename the name of the file to be downloaded
+     * @param destination the destination stream
+     */
+    void downloadToStreamByName(String filename, OutputStream destination);
+
+    /**
+     * Downloads the contents of the stored file specified by {@code filename} and by the revision in {@code options} and writes the
+     * contents to the {@code destination} Stream.
+     *
+     * @param filename the name of the file to be downloaded
+     * @param destination the destination stream
+     * @param options the download options
+     */
+    void downloadToStreamByName(String filename, OutputStream destination, GridFSDownloadByNameOptions options);
+
+    /**
+     * Finds all documents in the files collection.
+     *
+     * @return the GridFS find iterable interface
+     * @mongodb.driver.manual tutorial/query-documents/ Find
+     */
+    GridFSFindIterable find();
```

Comment:
In the CRUD spec we specifically decided that filters are mandatory.

Why this overload here?